crates/accelerate/src/two_qubit_decompose.rs

```diff
@@ -0,0 +1,190 @@
+// In numpy matrices real and imaginary components are adjacent:
+//   np.array([1,2,3], dtype='complex').view('float64')
+//   array([1., 0., 2., 0., 3., 0.])
+// The matrix faer::Mat<c64> has this layout.
+// faer::Mat<num_complex::Complex<f64>> instead stores a matrix
+// of real components and one of imaginary components.
+// In order to avoid copying we want to use `MatRef<c64>` or `MatMut<c64>`.
+
+use num_complex::Complex;
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::Python;
+use std::f64::consts::PI;
+
+use faer::Faer;
+use faer::IntoFaerComplex;
+use faer::{mat, Mat, MatRef, Scale};
+use faer_core::c64;
+use numpy::PyReadonlyArray2;
+
+use crate::utils;
+
+// FIXME: zero and one exist but I cant find the right incantation
+const C0: c64 = c64 { re: 0.0, im: 0.0 };
+const C1: c64 = c64 { re: 1.0, im: 0.0 };
+const C1IM: c64 = c64 { re: 0.0, im: 1.0 };
+
+// FIXME: Find a way to compute these matrices at compile time.
+fn transform_from_magic_basis(unitary: Mat<c64>) -> Mat<c64> {
+    let _b_nonnormalized: Mat<c64> = mat![
+        [C1, C1IM, C0, C0],
+        [C0, C0, C1IM, C1],
+        [C0, C0, C1IM, -C1],
+        [C1, -C1IM, C0, C0]
+    ];
+    let _b_nonnormalized_dagger = Scale(c64 { re: 0.5, im: 0.0 }) * _b_nonnormalized.adjoint();
+    _b_nonnormalized_dagger * unitary * _b_nonnormalized
+}
+
+// fn transform_to_magic_basis(unitary : Mat<c64>) -> Mat<c64> {
+//     let _b_nonnormalized : Mat<c64> = mat![[C1, C1IM, C0, C0],
+//                                            [C0, C0, C1IM, C1],
+//                                            [C0, C0, C1IM, -C1],
+//                                            [C1, -C1IM, C0, C0]];
+//     let _b_nonnormalized_dagger = Scale(c64 {re: 0.5, im: 0.0}) * _b_nonnormalized.adjoint();
+//     _b_nonnormalized * unitary * _b_nonnormalized_dagger
+// }
+
+// faer::c64 and num_complex::Complex<f64> are both structs
+// holding two f64's. But several functions are not defined for
+// c64. So we implement them here. These things should be contribute
+// upstream.
+
+pub trait PowF {
+    fn powf(self, pow: f64) -> c64;
+}
+
+impl PowF for c64 {
+    fn powf(self, pow: f64) -> c64 {
+        c64::from(Complex::<f64>::from(self).powf(pow))
```

Comment:
Seems a bit cleaner to call the built-in function for this: https://docs.rs/faer/latest/faer/complex_native/struct.c64.html#method.to_num_complex

```suggestion
        c64::from(self.to_num_complex().powf(pow))
```

Reply:
I missed that method, I think. I don't have a strong opinion, but `Complex::<f64>` seems more explicit to me.  You can't be sure from reading what `to_num_complex` converts to. If this is true then what use is `to_num_complex`?... We (I) can ask questions like this on the faer Discourse. They are quite responsive.